File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/core/NodeListImpl.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.core;

import java.util.Iterator;
import java.util.List;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

public class NodeListImpl implements NodeList {
	private List<Node> mList;

	public NodeListImpl(List<Node> aList) {
		mList = aList;
	}

	public int getLength() {
		return mList.size();
	}

	public Node item(int index) {
		return mList.get(index);
	}

	// gerekli mi?
	public Iterator GetEnumerator() {
		return mList.iterator(); // GetEnumerator();
	}

	public int getCount() {
		return mList.size();
	}
}

File: api-tsl/csharp/src/model/core/NodeListImpl.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.core
{
    public class NodeListImpl : XmlNodeList
    {
        private readonly IList<XmlNode> mList;

        public NodeListImpl(IList<XmlNode> aList)
        {
            mList = aList;
        }


        public virtual int Length
        {
            get
            {
                return mList.Count;
            }
        }

        public override XmlNode Item(int index)
        {
            return mList[index];
        }

        public override IEnumerator GetEnumerator()
        {
            return mList.GetEnumerator();
        }

        public override int Count
        {
            get
            {
                return mList.Count;
            }
        }
    }
}
